core/modules/qproc/geomAdapter.h

```diff
@@ -0,0 +1,105 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_QPROC_GEOMADAPTER_H
+#define LSST_QSERV_QPROC_GEOMADAPTER_H
+/**
+  * @file
+  *
+  * @brief Geometry adapter interface code
+  *
+  * @author Daniel L. Wang, SLAC
+  */
+
+// System headers
+#include <vector>
+
+// Other external headers
+#include "boost/make_shared.hpp"
+
+// LSST headers
+#include "sg/Box.h"
+#include "sg/Circle.h"
+#include "sg/Ellipse.h"
+#include "sg/ConvexPolygon.h"
+
+// Qserv headers
+#include "css/StripingParams.h"
+#include "qproc/QueryProcessingBug.h"
+
+
+namespace lsst {
+namespace qserv {
+namespace qproc {
+typedef double Coordinate;
+
+inline boost::shared_ptr<sg::Box>
+getBoxFromParams(std::vector<Coordinate> const& params) {
+    if(params.size() != 4) {
+        throw QueryProcessingBug("Invalid number or parameters for region");
+    }
+    return boost::make_shared<sg::Box>(sg::Box::fromDegrees(params[0], params[1], params[2], params[3]));
+}
+
+inline boost::shared_ptr<sg::Circle>
+getCircleFromParams(std::vector<Coordinate> const& params) {
+    // lon, lat radius_deg
+    if(params.size() != 3) {
+        throw QueryProcessingBug("Invalid number or parameters for region");
+    }
+    sg::LonLat center = sg::LonLat::fromDegrees(params[0], params[1]);
+    sg::Angle a = sg::Angle::fromDegrees(params[2]);
+    return boost::make_shared<sg::Circle>(sg::UnitVector3d(center), a);
+}
+
+inline boost::shared_ptr<sg::Ellipse>
+getEllipseFromParams(std::vector<Coordinate> const& params) {
+    // lon, lat, semimajang, semiminang, posangle
+    if(params.size() != 5) {
+        throw QueryProcessingBug("Invalid number or parameters for region");
+    }
+    sg::UnitVector3d uv3(sg::LonLat::fromDegrees(params[0], params[1]));
+    return boost::make_shared<sg::Ellipse>(
+        uv3,
+        sg::Angle::fromDegrees(params[2]),
+        sg::Angle::fromDegrees(params[3]),
+        sg::Angle::fromDegrees(params[4]));
+}
+
+inline boost::shared_ptr<sg::ConvexPolygon>
+getConvexPolyFromParams(std::vector<Coordinate> const& params) {
+    // polygon vertices, min 3 vertices, must get even number of params
+    if((params.size() <= 6) || ((params.size() & 1) != 0)) {
+        throw QueryProcessingBug("Invalid number or parameters for region");
+    }
+    std::vector<sg::UnitVector3d> uv3;
+    for(unsigned i=0; i < params.size(); i += 2) {
+        sg::LonLat vx = sg::LonLat::fromDegrees(params[i], params[i+1]);
+        uv3.push_back(sg::UnitVector3d(vx));
+    }
+    return boost::make_shared<sg::ConvexPolygon>(uv3);
+}
+
```

Comment:
In general, have you though about wrapping exceptions coming out of sg constructors in some way? Also, are you using `QueryProcessingBug` instead of some user-input related exception type because (some of?) the parameter length constraints should be enforced in the grammar?